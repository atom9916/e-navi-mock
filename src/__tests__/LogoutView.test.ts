/**
 * @jest-environment jsdom
 */

import { render, screen } from '@testing-library/vue'
import LogoutViewVue from '@/views/LogoutView.vue'
import { mount } from '@vue/test-utils'

describe('Logoutビューのテスト', () => {
  test('画面が正しく表示される', () => {
    render(LogoutViewVue)

    const button = screen.getByRole('button')

    expect(button).toHaveTextContent('再ログイン')
  })

  it('クリックするとログイン画面に遷移', async () => {
    const mockRouter = {
      push: jest.fn()
    }

    const wrapper = mount(LogoutViewVue, {
      global: {
        mocks: {
          $router: mockRouter
        }
      }
    })

    await wrapper.find('button').trigger('click')

    expect(mockRouter.push).toHaveBeenCalledTimes(1)
    expect(mockRouter.push).toHaveBeenCalledWith({ path: '/login' })
  })

  test('スナップショット', () => {
    const { container } = render(LogoutViewVue)
    expect(container).toMatchSnapshot()
  })
})
