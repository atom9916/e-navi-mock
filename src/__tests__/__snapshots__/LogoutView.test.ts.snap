// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logoutビューのテスト スナップショット 1`] = `
<div>
  
  <h2>
    ログアウトしました
  </h2>
  <div>
    <button>
      再ログイン
    </button>
  </div>
  
</div>
`;
